tools/ingest_csv.py

```diff
@@ -0,0 +1,195 @@
+    system_type = AssetType(name="System", uuid="00000000-0000-0000-0000-000000031302")
+    return [system_type, database_type, schema_type, table_type, column_type]
+
+
+def _validate_header(headers: List[str], csv_file: Path) -> List[int]:
+    index_fullname = [i for i, header in enumerate(headers) if header == "fullname"]
+    if len(index_fullname) != 2:
+        raise InvalidCSVException(f"The headers of the csv file {csv_file} do no match the required input.")
+
+    # there should be at least 1 node, parent and leaf
+    if index_fullname[0] < 3 or index_fullname[1] - index_fullname[0] < 5:
+        raise InvalidCSVException(f"The headers of the csv file {csv_file} do no match the required input.")
+
+    if headers[index_fullname[0] : index_fullname[0] + 2] + headers[index_fullname[1] :] != [
+        "fullname",
+        "domain_id",
+        "fullname",
+        "domain_id",
+        "source_code",
+        "highlights",
+        "transformation_display_name",
+    ]:
+        raise InvalidCSVException(f"The headers of the csv file {csv_file} do no match the required input.")
+
+    return index_fullname
+
+
+def _create_asset(
+    asset_types: List[str], asset_names: List[str], domain_id: str, fullname: str, csv_file: str, row: List[str]
+) -> Union[ParentAsset, LeafAsset]:
+    # Creating node asset
+    nodes = []
+    for asset_name, asset_type in zip(asset_names[:-2], asset_types[:-2]):
+        if asset_name:
+            nodes.append(Asset(name=asset_name, type=asset_type))
+    if not nodes:
+        raise InvalidCSVException(f"No nodes defined in {csv_file} in row {row}")
+    nodes_asset = NodeAsset(nodes=nodes)
+
+    # Creating the props when relevant
+    if fullname and domain_id:
+        props = AssetProperties(fullname=fullname, domain_id=domain_id)
+    else:
+        props = None
+
+    # Creating parrent asset
+    if not asset_names[-2]:
+        raise InvalidCSVException(f"Parent asset not defined in {csv_file} in row {row}")
+    parent_asset = ParentAsset(
+        nodes=nodes_asset.nodes, parent=Asset(name=asset_names[-2], type=asset_types[-2]), props=props
+    )
+
+    # Creating leaf asset - optionally
+    if asset_names[-1]:
+        return LeafAsset(
+            nodes=nodes_asset.nodes,
+            parent=parent_asset.parent,
+            leaf=Asset(name=asset_names[-1], type=asset_types[-1]),
+            props=props,
+        )
+
+    return parent_asset
+
+
+def _create_source_code(
+    source_code_text: str, highlights: str, transformation_display_name: str, custom_lineage_config: CustomLineageConfig
+) -> Optional[SourceCode]:
+    if not source_code_text:
+        return None
+
+    if highlights:
+        source_code_highlights = []
+        for highlight in highlights.split(","):
+            source_code_highlights.append(
+                SourceCodeHighLight(start=highlight.split(":")[0][1:], len=highlight.split(":")[1][:-1])
```

Comment:
Doing some string split here, shall we surround this with try...except in case highlights doesn't have the right format?